docusaurus/docs/how-to-guides/panel-plugins/plugin-internationalization.md

```diff
@@ -0,0 +1,264 @@
+---
+id: plugin-internationalization
+title: How to add translations to your plugin
+description: Translate your plugins
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - frontend
+  - tooling
+  - configuration
+  - translation
+  - localization
+  - internationalization
+  - webpack
+---
+
+# How to add translations to your plugin
+
+The Grafana UI is available in [several languages](https://grafana.com/docs/grafana/latest/administration/organization-preferences/#change-grafana-language). By default, plugins are available in English only and are not translated when you change your language settings. 
+
+Grafana uses the [i18next internationalization framework](https://www.i18next.com/) for translation purposes. If you want your plugin to be translatable to other languages you need to perform the changes described in this document.  
+
+## Before you begin
+
+:::info
+Available in Grafana >=v11.0.0 (US English) and Grafana >=v12.1.0 for the rest of languages.
+:::
+
+The following is recommended:
+
+* Basic knowledge of Grafana panel plugin development
+* Basic understanding of the [i18next internationalization framework](https://www.i18next.com/)
+
+## Overview of the files affected by translation
+
+If you create your plugin running the `create-plugin` scaffolding tool you'll get the following folder layout and key file:
+
+```
+myorg-myplugin-plugintype/
+├── pkg/
+│   ├── main.go
+│   └── plugin/
+├── src/
+│   ├── module.ts
+│   ├── plugin.json
+└── tests/
+├── CHANGELOG.md
+├── docker-compose.yaml
+├── go.mod
+├── package.json
+├── LICENSE
+├── Magefile.go
+├── README.md
+```
+
+:::note
+
+Since `create-plugin` is constantly being improved your folder structure could look slightly different.
+
+:::
+
+Enabling plugin translation involves updating the following files:
+
+- `docker-compose.yaml` 
+- `plugin.json`
+- `module.ts`
+- `webpack.config.ts`
+- `package.json`
+
+## Set up your plugin for translation
+
+Follow these steps to update your plugin and set it up for translation:
+
+### Enable translation in your Grafana instance 
+
+To translate your plugin you need to enable the feature toggle `localizationForPlugins` on your Grafana instance.
+
+To do so, update `docker-compose.yaml` in your plugin:
+
+```yaml title="docker-compose.yaml"
+services:
+  grafana:
+    extends:
+      file: .config/docker-compose-base.yaml
+      service: grafana
+    environment:
+      GF_FEATURE_TOGGLES_ENABLE: localizationForPlugins
+```
+
+### Define the languages and Grafana dependencies
+
+Set up the translation languages for your plugin. 
+
+To do so, insert a `languages` section into the `plugin.json` file of the plugin. For example, if you want to add US English and Brazilian Portuguese:
```

Comment:
I think we need to emphasize the `grafanaDependency` change again, super important because this is probably a breaking change for many plugins.